--- /Users/consi/Downloads/Musashi-master/m68kconf.h	2016-04-28 09:41:56.000000000 +0200
+++ m68kconf.h	2016-05-07 20:32:03.000000000 +0200
@@ -52,7 +52,7 @@
  * to OPT_ON and use m68kmame.h to configure the 68k core.
  */
 #ifndef M68K_COMPILE_FOR_MAME
-#define M68K_COMPILE_FOR_MAME      OPT_ON
+#define M68K_COMPILE_FOR_MAME      OPT_OFF
 #endif /* M68K_COMPILE_FOR_MAME */
 
 
@@ -131,8 +131,8 @@
 /* If ON, CPU will call the instruction hook callback before every
  * instruction.
  */
-#define M68K_INSTRUCTION_HOOK       OPT_OFF
-#define M68K_INSTRUCTION_CALLBACK() your_instruction_hook_function()
+#define M68K_INSTRUCTION_HOOK       OPT_SPECIFY_HANDLER
+#define M68K_INSTRUCTION_CALLBACK() m68k_instr_callback()
 
 
 /* If ON, the CPU will emulate the 4-byte prefetch queue of a real 68000 */
