--- vfprintf.c.orig	2017-09-24 11:35:58.000000000 +0200
+++ vfprintf.c	2017-09-24 11:43:39.000000000 +0200
@@ -5,7 +5,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <stdarg.h>
-#include <unistd.h>
+#include <proto/dos.h>
 
 #define BUFFER_SIZE	32768
 
@@ -22,5 +22,5 @@
 	if (rv > BUFFER_SIZE - 1)
 		rv = BUFFER_SIZE - 1;
 
-	return _fwrite(buffer, rv, file);
+    return Write((BPTR) file->_fileno, buffer, rv);
 }
